src/main/java/heronarts/lx/midi/surface/APC40Mk2.java

```diff
@@ -381,12 +504,19 @@ private void unregister() {
       }
       if (this.device != null) {
         for (int i = 0; i < this.knobs.length; ++i) {
-          if (this.knobs[i] != null) {
+          if (this.knobs[i] == null) {
+            continue;
+          } else if (this.knobs[i] instanceof AggregateParameter) {
+            AggregateParameter ap = (AggregateParameter) this.knobs[i];
+            for (LXListenableParameter sub : ap.subparameters.values()) {
```

Comment:
Per the above comments about getRemoteControls() - there is no guarantee here that we have registered as a listener to every sub-parameter of an `AggregateParameter`, they may or may not have all been returned by getRemoteControls().

I think the safest solution here is that we should only listen to parameters directly registered as knobs, and we need to make sure that any `AggregateParameter` does properly fire a listener message when something which changes its display has happened.

Otherwise I worry that this extra-listener management could get brittle.

Reply:
I think I've addressed all of your feedback except for this one. I think I need a bit more detail if I'm going to make this change to AggregateParameter.

Reply:
Done